libtenzir/src/tql2/ast.cpp

```diff
@@ -169,6 +170,51 @@ auto to_operand(const ast::expression& x) -> std::optional<operand> {
     });
 }
 
+// @brief Optimize x > now() +- $y <=> x > $now +- $y and x > now() +- $y
+auto optimize_now(const ast::expression& left,
+                  const relational_operator& rel_op,
+                  const ast::expression& right) -> std::optional<expression> {
```

Comment:
It looks like this function only works for a small subset of `relational_operator`. Can you either adjust its signature or add an assertion for this?